lib/PuppeteerSharp/Helpers/TempDirectory.cs

```diff
@@ -31,31 +31,20 @@ public TempDirectory(string path)
 
         ~TempDirectory()
         {
-            Dispose(false);
+            DisposeCore();
         }
 
         public string Path { get; }
 
         public void Dispose()
         {
             GC.SuppressFinalize(this);
-            Dispose(true);
+            DisposeCore();
         }
 
         public override string ToString() => Path;
 
-        public Task DeleteAsync(CancellationToken cancellationToken = default)
-            => _deleteTask ?? (_deleteTask = DeleteAsync(Path, CancellationToken.None));
-
-        protected virtual void Dispose(bool disposing)
-        {
-            if (_deleteTask == null)
-            {
-                _ = DeleteAsync();
-            }
-        }
-
-        private static async Task DeleteAsync(string path, CancellationToken cancellationToken = default)
+        private static async Task DeleteAsync(string path)
```

Comment:
`DeleteAsync(CancellationToken cancellationToken = default)` has never been used outside this class since it was introduced in #585, so `cancellationToken` is always `CancellationToken.None`.

I don't have the full overview of what #585 does, so I refrained from removing the fire-and-forget invocation of of `DeleteAsync`.